File: Examen Parcial/20221309_EX1/Pregunta03/TransitSoft/TransitSoftBusiness/VehiculoBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TransitSoftModel;
using TransitSoftPersistance.DAO;
using TransitSoftPersistance.DAOImpl;

namespace TransitSoftBusiness
{
    public class VehiculoBO
    {
        private VehiculoDAO vehiculoDAO;

        public VehiculoBO()
        {
            this.vehiculoDAO = new VehiculoDAOImpl();
        }

        public int Insertar(string placa, string marca, string modelo, int anho, int propietarioId)
        {
            VehiculoDTO vehiculoDTO = new VehiculoDTO();
            vehiculoDTO.Placa = placa;
            vehiculoDTO.Marca = marca;
            vehiculoDTO.Modelo = modelo;
            vehiculoDTO.Anho = anho;
            vehiculoDTO.Propietario = new PropietarioDTO();
            vehiculoDTO.Propietario.Id = propietarioId;
            return this.vehiculoDAO.Insertar(vehiculoDTO);
        }

        public int Eliminar(int id)
        {
            VehiculoDTO vehiculoDTO = new VehiculoDTO();
            vehiculoDTO.Id = id;
            return this.vehiculoDAO.Eliminar(vehiculoDTO);
        }

        public int Modificar(int id, string placa, string marca, string modelo, int anho, int propietarioId)
        {
            VehiculoDTO vehiculoDTO = new VehiculoDTO();
            vehiculoDTO.Id = id;
            vehiculoDTO.Placa = placa;
            vehiculoDTO.Marca = marca;
            vehiculoDTO.Modelo = modelo;
            vehiculoDTO.Anho = anho;
            vehiculoDTO.Propietario = new PropietarioDTO();
            vehiculoDTO.Propietario.Id = propietarioId;
            return this.vehiculoDAO.Modificar(vehiculoDTO);
        }

        public VehiculoDTO ObtenerPorId(int id)
        {
            return this.vehiculoDAO.ObtenerPorId(id);
        }

        public BindingList<VehiculoDTO> ListarTodos()
        {
            return this.vehiculoDAO.ListarTodos();
        }
    }
}

File: EX2/Conexion RESTful Transitsoft/ServidorRESTFul/TransitSoftBusiness/src/main/java/pe/edu/pucp/transitsoft/bo/server/VehiculoBO.java
package pe.edu.pucp.transitsoft.bo.server;

import java.util.ArrayList;
import pe.edu.pucp.transitsoft.dao.VehiculoDAO;
import pe.edu.pucp.transitsoft.daoImp.VehiculoDAOImpl;
import pe.edu.pucp.transitsoft.model.VehiculoDTO;

public class VehiculoBO {

    private final VehiculoDAO vehiculoDAO;
    
    public VehiculoBO(){
        this.vehiculoDAO = new VehiculoDAOImpl();
    }
    
    public Integer insertar(VehiculoDTO vehiculoDTO){
        return this.vehiculoDAO.insertar(vehiculoDTO);
    }
    
    public VehiculoDTO obtenerPorId(Integer id){
        return this.vehiculoDAO.obtenerPorId(id);
    }
    
    public ArrayList<VehiculoDTO> listarTodos(){
        return this.vehiculoDAO.listarTodos();
    }
    
    public Integer modificar(VehiculoDTO vehiculoDTO){
        return this.vehiculoDAO.modificar(vehiculoDTO);
    }
    
    public Integer eliminar(Integer id){
        VehiculoDTO vehiculoDTO = new VehiculoDTO();
        vehiculoDTO.setId(id);
        return this.vehiculoDAO.eliminar(vehiculoDTO);
    }
}
